ludwig/models/new_ludwig_model.py

```diff
@@ -719,6 +721,109 @@ def evaluate(
+            skip_save_test_predictions=False,  # skipcq: PYL-W0613
+            skip_save_test_statistics=False,  # skipcq: PYL-W0613
+            output_directory='results',
+            gpus=None,
+            gpu_memory_limit=None,
+            allow_parallel_threads=True,
+            use_horovod=None,
+            random_seed=default_random_seed,
+            debug=False,
+            **kwargs
+    ):
+        (
+            train_stats,
+            preprocessed_data
+        ) = self.train(
+            dataset=dataset,
+            training_set=training_set,
+            validation_set=validation_set,
+            test_set=test_set,
+            training_set_metadata=training_set_metadata,
+            data_format=data_format,
+            experiment_name=experiment_name,
+            model_name=model_name,
+            model_load_path=model_load_path,
+            model_resume_path=model_resume_path,
+            skip_save_training_description=skip_save_training_description,
+            skip_save_training_statistics=skip_save_training_statistics,
+            skip_save_model=skip_save_model,
+            skip_save_progress=skip_save_progress,
+            skip_save_log=skip_save_log,
+            skip_save_processed_input=skip_save_processed_input,
+            output_directory=output_directory,
+            gpus=gpus,
+            gpu_memory_limit=gpu_memory_limit,
+            allow_parallel_threads=allow_parallel_threads,
+            use_horovod=use_horovod,
+            random_seed=random_seed,
+            debug=debug,
+        )
+
+        (_,  # training_set
+         _,  # validation_set
+         test_set,
+         training_set_metadata) = preprocessed_data
+
+        if test_set is not None:
+            if self.model_definition[TRAINING]['eval_batch_size'] > 0:
+                batch_size = self.model_definition[TRAINING]['eval_batch_size']
+            else:
+                batch_size = self.model_definition[TRAINING]['batch_size']
+
+            # todo tf2 refactor: figure out where this goes given NewLudwigModel
+            # if a model was saved on disk, reload it
+            # model_dir = os.path.join(self.exp_dir_name, 'model')
+            # if is_model_dir(model_dir):
+            #     model = NewLudwigModel.load(model_dir,
+            #                          use_horovod=use_horovod,
+            #                          gpus=gpus,
+            #                          gpu_memory_limit=gpu_memory_limit,
+            #                          allow_parallel_threads=allow_parallel_threads)
+
+            # predict
+            test_results = self.evaluate(
+                test_set,
+                data_format=data_format,
+                batch_size=batch_size,
+                collect_predictions=True,
+                collect_overall_stats=True,
+                debug=debug
+            )
+        else:
+            test_results = None
+
+        return (
```

Comment:
The caller of this function already has a reference to the NewLudwigModel object, so I believe that returning self, self.exp_dir_name and self.model_definition are superfluous.
I would return only:
```
test_results,
train_stats,
preprocessed_data
```

Reply:
done.